index.js

```diff
@@ -10,10 +10,17 @@
 var typeOf = require('kind-of');
 
 module.exports = function isNumber(num) {
+  if (!num && num !== 0) return false;
+
   var type = typeOf(num);
   if (type !== 'number' && type !== 'string') {
     return false;
   }
+  if (type === 'string') {
+    num = num.trim();
+    if (!num.length) return false;
+  }
+
   var n = +num;
```

Comment:
Hm. It's even better if we remove 19-23 and add

```js
if (n === 0 && type === 'string' && !num.trim()) return false;
```
below `var n = +num`

Not sure enough why but... tests are still passing.